sdk/Trace/ExporterFactory.php

```diff
@@ -0,0 +1,165 @@
+
+    public function __construct(string $name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+      * Returns the coresponding Exporter via the configuration string
+      *
+      * @param string $configurationString String containing unextracted information for Exporter creation
+      * Should follow the format: contribType+baseUrl?option1=a
+      * Query string is optional and based on the Exporter
+      */
+    public function fromConnectionString(string $configurationString)
+    {
+        $strArr = explode('+', $configurationString);
+        // checks if input is given with the format type+baseUrl
+        if (sizeof($strArr) != 2) {
+            return null;
+        }
+
+        $contribName = strtolower($strArr[0]);
+        $endpointUrl = $strArr[1];
+
+        if (!$this->isAllowed($contribName)) {
+            return null;
+        }
+
+        // endpoint is only parsed if it is provided
+        $dsn = empty($endpointUrl) ? '' : DsnParser::parse($endpointUrl);
+        $endpointUrl = $this->parseBaseUrl($dsn);
+        // parameters are only retrieved if there was an endpoint given
+        $args = empty($dsn) ? [] : $dsn->getParameters();
+        
+        switch ($contribName) {
+            case 'jaeger':
+                return $exporter = $this->generateJaeger($endpointUrl);
+            case 'zipkin':
+                return $exporter = $this->generateZipkin($endpointUrl);
+            case 'newrelic':
+                return $exporter = $this->generateNewrelic($endpointUrl, $args['licenseKey'] ?? null);
+            case 'otlp':
+                return $exporter = $this->generateOtlp();
+            case 'otlpgrpc':
+                return $exporter = $this->generateOtlpGrpc();
+            case 'zipkintonewrelic':
+                return $exporter = $this->generateZipkinToNewrelic($endpointUrl, $args['licenseKey'] ?? null);
+            }
+    }
+
+    private function isAllowed(string $exporter)
+    {
+        return array_key_exists($exporter, $this->allowedExporters) && $this->allowedExporters[$exporter];
+    }
+
+    // constructs the baseUrl with the arguments retrieved from the raw baseUrl
+    private function parseBaseUrl($dsn)
+    {
+        if ($dsn == false) {
+            return null;
+        }
+        $parsedUrl = '';
+        $parsedUrl .= empty($dsn->getScheme()) ? '' : $dsn->getScheme() . '://';
+        $parsedUrl .= empty($dsn->getHost()) ? '' : $dsn->getHost();
+        $parsedUrl .= empty($dsn->getPort()) ? '' : ':' . $dsn->getPort();
+        $parsedUrl .= empty($dsn->getPath()) ? '' : $dsn->getPath();
+
+        return $parsedUrl;
+    }
+
+    private function generateJaeger(string $endpointUrl)
+    {
+        $exporter = new JaegerExporter(
+            $this->name,
+            $endpointUrl,
+            new Client(),
+            new HttpFactory(),
+            new HttpFactory()
```

Comment:
Is there a reason we're generating 2 `HttpFactory` instances here?  This happens a couple of times further in the PR as well.

Reply:
I was basing this off of the examples under the example folder. 
The constructor for this is constructed as such:
`   public function __construct(
        $name,
        string $endpointUrl,
        ClientInterface $client,
        RequestFactoryInterface $requestFactory,
        StreamFactoryInterface $streamFactory,
        SpanConverter $spanConverter = null
    )`
 Would only one `HttpFactory` suffice ?

Reply:
Only one instance, it can be constructed only once, yes.

Reply:
I think both Arguments refer to different Interface implementations.  One is a RequestFactory and the other is a StreamFactory, the HttpFactory implements both, hence the reuse.

Reply:
Correct, it's a side effect of the specific implementation, but the argument types are correct.